stock_available_unreserved/models/product.py

```diff
@@ -0,0 +1,141 @@
+        string='Unreserved stock quantity')
+
+    @api.multi
+    def _compute_product_available_not_res(self):
+        no_new = self.filtered(
+            lambda x: not isinstance(x.id, models.NewId))
+        res = no_new._product_available()
+        for tmpl in no_new:
+            qty = res[tmpl.id]['qty_available_not_res']
+            tmpl.qty_available_not_res = qty
+            text = res[tmpl.id]['qty_available_stock_text']
+            tmpl.qty_available_stock_text = text
+
+    @api.multi
+    def _product_available(self, name=None, arg=False):
+        prod_available = super(ProductTemplate, self)._product_available(name,
+                                                                         arg)
+
+        variants = self.env['product.product']
+        for product in self:
+            variants += product.product_variant_ids
+        variant_available = variants._product_available()
+
+        for product in self:
+            if isinstance(product.id, models.NewId):
+                continue
+            qty_available_not_res = 0.0
+            text = ''
+            for p in product.product_variant_ids:
+                qty = variant_available[p.id]["qty_available_not_res"]
+                qty_available_not_res += qty
+                text = variant_available[p.id]["qty_available_stock_text"]
+            prod_available[product.id].update({
+                "qty_available_not_res": qty_available_not_res,
+                "qty_available_stock_text": text,
+            })
+        return prod_available
+
+    @api.multi
+    def action_open_quants_unreserved(self):
+        products = self.mapped('product_variant_ids').ids
+        result = self.env.ref('stock.product_open_quants').read()[0]
+        result['domain'] = "[('product_id','in',[" + ','.join(
+            map(str, products)) + "]), ('reservation_id', '=', False)]"
+        result[
+            'context'] = "{'search_default_locationgroup': 1, " \
+                         "'search_default_internal_loc': 1}"
+        return result
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    qty_available_not_res = fields.Float(
+        string='Qty Available Not Reserved', digits=UNIT,
+        compute='_compute_qty_available_not_res')
+
+    qty_available_stock_text = fields.Char(
+        compute='_compute_qty_available_not_res', string='Available per stock')
+
+    @api.multi
+    def _compute_qty_available_not_res(self):
+        res = self._product_available()
+        for prod in self:
+            qty = res[prod.id]['qty_available_not_res']
+            text = res[prod.id]['qty_available_stock_text']
+            prod.qty_available_not_res = qty
+            prod.qty_available_stock_text = text
+
+    @api.model
+    def _prepare_domain_available_not_res(self, products):
+        domain_products = [('product_id', 'in', products.mapped('id'))]
+        domain_quant = []
+        domain_quant_loc, _, _ = products._get_domain_locations()
```

Comment:
@StefanRijnhart  Be carefull by using `_` as variable you hide  `from odoo import api, fields, models, _`